src/components/Filters/FilterDropdownMenu.tsx

```diff
@@ -0,0 +1,198 @@
+            <SelectField
+              label="Group by"
+              compact
+              labelStyle="inline"
+              sizeToContent
+              options={groupBy.options}
+              getOptionValue={(o) => o.id}
+              getOptionLabel={(o) => o.name}
+              value={groupBy.value}
+              onSelect={(g) => g && groupBy.setValue(g)}
+            />
+          )}
+
+          {/* Render all filters (non-checkbox first, then checkbox) */}
+          {renderFilters()}
+
+          {/* Clear button at end of filter controls */}
+          {activeFilterCount > 0 && (
+            <Button label="Clear" variant="tertiary" onClick={() => onChange({} as F)} {...testId.clearBtn} />
+          )}
+        </div>
+      )}
+
+      {/* Filter chips (and clear button) shown when dropdown is closed */}
+      {!isOpen && (
+        <FilterChips
+          filter={filter}
+          filterImpls={filterImpls}
+          onChange={onChange}
+          onClear={() => onChange({} as F)}
+          testId={testId}
+        />
+      )}
+    </>
+  );
+}
+
+interface FilterChipsProps<F extends Record<string, unknown>> {
+  filter: F;
+  filterImpls: ReturnType<typeof buildFilterImpls<F>>;
+  onChange: (filter: F) => void;
+  onClear: () => void;
+  testId: ReturnType<typeof useTestIds>;
+}
+
+function FilterChips<F extends Record<string, unknown>>({
+  filter,
+  filterImpls,
+  onChange,
+  onClear,
+  testId,
+}: FilterChipsProps<F>) {
+  const chips = safeEntries(filterImpls).flatMap(([key]) => {
+    const value = filter[key];
+    if (!isDefined(value)) return [];
+
+    if (Array.isArray(value)) {
+      return value.map((item) => {
+        const chipKey = `${String(key)}_${item}`;
+        const newArray = value.filter((v) => v !== item);
+        return (
+          <ToggleChip
+            key={chipKey}
+            text={titleCase(String(item))}
+            onClick={() => onChange(updateFilter(filter, key, newArray.length > 0 ? (newArray as any) : undefined))}
+            {...testId[`chip_${chipKey}`]}
+          />
+        );
+      });
+    }
+
+    return (
+      <ToggleChip
+        key={String(key)}
+        text={titleCase(String(value))}
+        onClick={() => onChange(updateFilter(filter, key, undefined))}
+        {...testId[`chip_${String(key)}`]}
+      />
+    );
+  });
+
+  if (chips.length === 0) return null;
+
+  return (
+    <div css={Css.df.gap1.fww.aic.order(1).w100.$}>
+      {chips}
+      <Button label="Clear" variant="tertiary" onClick={onClear} {...testId.clearBtn} />
+    </div>
+  );
+}
+
+/** Convert FilterDefs to FilterImpls by evaluating the factory functions */
+function buildFilterImpls<F extends Record<string, unknown>>(filterDefs: FilterDefs<F>): FilterImpls<F> {
+  return Object.fromEntries(safeEntries(filterDefs).map(([key, fn]) => [key, fn(key as string)])) as FilterImpls<F>;
+}
+
+/** Capitalize the first letter of each word */
+function titleCase(str: string): string {
```

Comment:
was surprised we're not using `lodash` here, so I just built this helper - I could add it, but wasn't sure